src/lm_buddy/jobs/evaluation/hf_evaluate.py

```diff
@@ -0,0 +1,115 @@
+"""
+lm-buddy entrypoint to run summary evaluation using huggingface eval
+"""
+
+import json
+import time
+from pathlib import Path
+
+import s3fs
+from loguru import logger
+from tqdm import tqdm
+
+from lm_buddy.configs.jobs.hf_evaluate import HuggingFaceEvalJobConfig
+from lm_buddy.configs.vllm import VLLMCompletionsConfig
+from lm_buddy.constants import LM_BUDDY_RESULTS_PATH
+from lm_buddy.jobs.asset_loader import (
+    HuggingFaceDatasetLoader,
+    HuggingFaceModelLoader,
+)
+from lm_buddy.jobs.common import EvaluationResult
+from lm_buddy.jobs.evaluation.metrics import EvaluationMetrics
+from lm_buddy.jobs.model_clients import (
+    HuggingFaceModelClient,
+    OpenAIModelClient,
+    PipelineModelClient,
+)
+
+
+def save_outputs(config: HuggingFaceEvalJobConfig, evaluation_results: dict) -> Path:
+    storage_path = config.evaluation.storage_path
+
+    # generate local temp file ANYWAY
+    # (we don't want to lose all eval data if there is an issue wth s3)
+    local_path = Path(LM_BUDDY_RESULTS_PATH) / config.name / "eval_results.json"
+    local_path.parent.mkdir(exist_ok=True, parents=True)
+    with local_path.open("w") as f:
+        json.dump(evaluation_results, f)
+
+    # copy to s3 and return path
+    if storage_path is not None and storage_path.startswith("s3://"):
+        s3 = s3fs.S3FileSystem()
+        if storage_path.endswith("/"):
+            storage_path = "s3://" + str(Path(storage_path[5:]) / config.name / "eval_results.json")
+        logger.info(f"Storing into {storage_path}...")
+        s3.put_file(local_path, storage_path)
+        return storage_path
+    else:
+        return local_path
+
+
+def run_eval(config: HuggingFaceEvalJobConfig) -> Path:
```

Comment:
should the return type be a list of Paths?

Reply:
My thought was one model + one evaluation (w multiple metrics) => one output path.

Were you referring to the possibility of evaluating multiple models? My rationale for keeping it atomic is that we'll likely want to parallelize evals on different models with different ray jobs, thus the logic for splitting the higher-level task in multiple evals (and also for retrieving different eval results and presenting them together) should be in the mzai-platform backend. WDYT?

Reply:
ah yep, my bad! here mostly I was referring to this return `return save_outputs(config, evaluation_results)` which I thought returned both paths.

Reply:
Ah that makes sense! Yeah, the local one for me was mostly a temp file for the s3 one. It definitely comes useful for local testing, but will eventually be lost when ray jobs run lm-buddy